04/dbernhard/arr_of_string.oat

```diff
@@ -0,0 +1,6 @@
+int program (int argc, string[] argv) {
+  /* 0 argument is the name of the program */
+  var arr = array_of_string(argv[1]);
+  
+  return arr[1];
```

Comment:
Not sure about this, I know that in our LLVM representation we use only i64, but the actual type of a string character is an i8 array. At the Oat level, I think the typechecker would complain about. I know we are developing an unsafe Oat version, but I whink it is still better to include only well-formed programs since if I remember properly from the moodle discussions they told us that they ara not gonna feed bad programns for testing. Also well-formed programs have to be accepted by everyone compiler, instead bad ones may be rejected or not.

What do you think about?

Reply:
I'd argue that this is indeed a valid program. The built in function array_of_string accepts a reference to an RString and returns a reference to RArray of TInt which we can index into.

Reply:
Oh sorry you are right I missed the type definition of the array_of_string function